models/branch.go

```diff
@@ -0,0 +1,43 @@
+package models
+
+import (
+	"github.com/gogits/git-module"
+)
+
+type Branch struct {
+	Path		string
+	Name      	string
+}
+
+func GetBranchesByPath(path string) ([]*Branch, error) {
+	gitRepo, err := git.OpenRepository(path)
+	if err != nil {
+		return nil, err
+	}
+
+	brs, err := gitRepo.GetBranches()
+	if err != nil {
+		return nil, err
+	}
+
+	Branches := make([]*Branch, len(brs))
+	for i := range brs {
+		Branches[i] = &Branch{
+			Path: path,
+			Name: brs[i],
+		}
+	}
+	return Branches, nil
+}
+
+func GetBranchesByRepo(user,repo string) ([]*Branch, error) {
```

Comment:
That an old artefact from a first try but after i have found that ctx.Repo.Repository is prefill under api so i use a direct call ctx.Repo.Repository.GetBranches(). Should i comment it or remove it completely ?